Quaternion/Quaternion.py

```diff
@@ -275,38 +326,35 @@ def _quat2equatorial(self):
         """
         Determine Right Ascension, Declination, and Roll for the object quaternion
 
-        :returns: RA, Dec, Roll
+        :returns: N x (RA, Dec, Roll)
         :rtype: numpy array [ra,dec,roll]
         """
 
-        q = self.q
-        q2 = self.q**2
+        q = np.atleast_2d(self.q)
+        q2 = q ** 2
 
         # calculate direction cosine matrix elements from $quaternions
-        xa = q2[0] - q2[1] - q2[2] + q2[3]
-        xb = 2 * (q[0] * q[1] + q[2] * q[3])
-        xn = 2 * (q[0] * q[2] - q[1] * q[3])
-        yn = 2 * (q[1] * q[2] + q[0] * q[3])
-        zn = q2[3] + q2[2] - q2[0] - q2[1]
+        xa = q2[..., 0] - q2[..., 1] - q2[..., 2] + q2[..., 3]
+        xb = 2 * (q[..., 0] * q[..., 1] + q[..., 2] * q[..., 3])
+        xn = 2 * (q[..., 0] * q[..., 2] - q[..., 1] * q[..., 3])
+        yn = 2 * (q[..., 1] * q[..., 2] + q[..., 0] * q[..., 3])
+        zn = q2[..., 3] + q2[..., 2] - q2[..., 0] - q2[..., 1]
 
         # Due to numerical precision this can go negative.  Allow *slightly* negative
         # values but raise an exception otherwise.
         one_minus_xn2 = 1 - xn**2
-        if one_minus_xn2 < 0:
-            if one_minus_xn2 < -1e-12:
+        if np.any(one_minus_xn2 < 0):
+            if np.any(one_minus_xn2 < -1e-12):
                 raise ValueError('Unexpected negative norm: {}'.format(one_minus_xn2))
-            one_minus_xn2 = 0
+            one_minus_xn2[one_minus_xn2 < 0] = 0
 
         # ; calculate RA, Dec, Roll from cosine matrix elements
-        ra = degrees(atan2(xb, xa))
-        dec = degrees(atan2(xn, sqrt(one_minus_xn2)))
-        roll = degrees(atan2(yn, zn))
-        if (ra < 0):
-            ra += 360
-        if (roll < 0):
-            roll += 360
-
-        return np.array([ra, dec, roll])
+        ra = np.degrees(np.arctan2(xb, xa))
+        dec = np.degrees(np.arctan2(xn, np.sqrt(one_minus_xn2)))
+        roll = np.degrees(np.arctan2(yn, zn))
+        ra[ra < 0] = ra[ra < 0] + 360
+        roll[roll < 0] = roll[roll < 0] + 360
+        return np.moveaxis(np.array([ra, dec, roll]), 0, -1)
```

Comment:
Add a comment explaining what's happening here.  Not totally obvious to me.

Reply:
I will add this:
```
        # all negative angles are incremented by 360,
        # the output is in the (0,360) interval instead of in (-180, 180)
```

Reply:
I was referring to the `moveaxis`.

Reply:
ah! right. Not trivial.

It is a transpose. The resulting ra, dec roll have shape (N,) (where N is an arbitrary tuple), then they are stacked so the result is (3,N), moving the axis produces (N,3).

A transpose of (3,2,5) would produce (5,2,3) which is not what we want: (2,5,3).

Perhaps it is better to just create the result array first, and then fill it at the right position. No need to transpose then.

Reply:
With the comment it is clear enough, but true that creating / filling in the output would be easier to read and not *need* a comment.  Either way is fine.